### ISSUE TITLE
Custom form fields don't have required attributes

### ISSUE THREAD

**Issue by abirhoss** [impress-org/givewp#5777]
Hi team,

I have found an issue when adding custom form fields using the Form Field Manager add-on that's related to the `validateLineItem( $array )` function on line 145 in `wp-content/plugins/give/src/Receipt/Section.php`

## User Story

As a donor, I want donate so that I can help with a cause.

## Details

When creating custom form fields using the `Form Field Manager` add-on, if the `label` is left empty, it leads to a 500 error when a donor has completed a donation and the form tries to generate the donation receipt.

The donor just sees an indefinite spinning loader

## Expected Behavior

The donor should be able to see their donation receipt instead of an indefinite loader

## Steps to Reproduce

1. Turn on the Form Field Manager add on plugin
2. Go to a donation form that is using the multi-step template
2. Scroll down to the Form Field Manager section
3. Select `Checkbox`
4. Enter a `Meta Key` and at least one checkbox option
5. Leave `Form Field Location` to default `Above all fields`
5. Click `Update` to save form
6. Make a test donation with the field checked

![Screen Shot 2021-04-03 at 5 13 20 pm](https://user-images.githubusercontent.com/25050149/113470814-1272e880-94a4-11eb-9793-30edaead5696.png)

## Visuals

![Screen Shot 2021-04-03 at 5 48 10 pm](https://user-images.githubusercontent.com/25050149/113470955-e0ae5180-94a4-11eb-8e70-9f64d2dd0328.png)

## Additional Context

The `validateLineItem( $array )` function validates that each donation form field has an `id`, `label` and `value` when generating a receipt. 

```
protected function validateLineItem( $array ) {
    $required = [ 'id', 'label', 'value' ];
    $array    = array_filter( $array ); // Remove empty values.

    if ( array_diff( $required, array_keys( $array ) ) ) {
      throw new InvalidArgumentException(
        esc_html__(
          'Invalid receipt section line item. Please provide valid line item id, label, and value.',
          'give'
        )
      );
    }
  }
```

This is fine, but this attribute requirement is not documented anywhere nor enforced when creating custom form fields using the `Form Field Manager` addon when editing donation forms. 

I think enforcing these field atrributes at the receipt generation stage is too late in the call stack, this requirement should be brought up to the admin interface when creating custom form fields.

In fact, none of the attributes within a custom form field are required. As a result, if a custom field has an empty `label`, it leads to 500 error when trying to generate the receipt after a donation is completed

On the site error logs, I see the following error:

```
"PHP message: PHP Fatal error: Uncaught InvalidArgumentException: 
Invalid receipt section line item. Please provide a valid line item ID, label, and value.
in /.../public/wp-content/plugins/give/src/Receipt/Section.php:150

Stack trace:

#0 /.../public/wp-content/plugins/give/src/Receipt/Section.php(89): Give\Receipt\Section->validateLineItem(Array)

#1 /.../public/wp-content/plugins/give-form-field-manager/src/Receipt/UpdateDonationReceipt.php(41): Give\Receipt\Section->addLineItem(Array)

#2 /.../public/wp-content/plugins/give-form-field-manager/includes/functions.php(294): GiveFormFieldManager\Receipt\UpdateDonationReceipt->apply()

#3 /.../public/wp-includes/class-wp-hook.php(292): give_ffm_register_line_items(Object(Give\Receipt\DonationReceipt))

#4 /.../public/wp-includes/class-wp-hook.php(316): WP_Hook->apply_filters(NULL, Array)

#5 while reading response header from upstream, ..., request: "GET /wp-admin/admin-ajax.php?action=get_receipt&shortcode_atts=&donation_id=&receipt_type=&wp-give_session_... HTTP/1.0", upstream: "...", host: "...", referrer: "https://.../thank-you/"
```

A missing `Meta Key` field attribute also prevents the submitted custom field data from persisting in the database, but I have opened another issue for that #

This issue affects all pages where a receipt would be shown, such as the new Donor Dashboard which is how I found the issue:

![Screen Shot 2021-04-03 at 4 28 27 pm](https://user-images.githubusercontent.com/25050149/113471083-c032c700-94a5-11eb-8f30-923c7647bccc.png)

## Sample Field Attributes

If function `validateLineItem( $array )` receives the following array field as an argument

```
array(12) {
  ["input_type"]=>
  string(8) "checkbox"
  ["template"]=>
  string(14) "checkbox_field"
  ["required"]=>
  string(2) "no"
  ["label"]=>
  string(0) ""
  ["name"]=>
  string(18) "this_is_a_checkbox"
  ["is_meta"]=>
  string(3) "yes"
  ["field_width"]=>
  string(4) "full"
  ["css"]=>
  string(0) ""
  ["help"]=>
  string(0) ""
  ["options"]=>
  array(1) {
    [0]=>
    string(18) "This is a checkbox"
  }
  ["value"]=>
  string(18) "This is a checkbox"
  ["id"]=>
  string(18) "this_is_a_checkbox"
}
```

It proceeds to remove all empty fields using `$array = array_filter( $array );`. This results in an array that looks like below:

```
array(9) {
  ["input_type"]=>
  string(8) "checkbox"
  ["template"]=>
  string(14) "checkbox_field"
  ["required"]=>
  string(2) "no"
  ["name"]=>
  string(18) "this_is_a_checkbox"
  ["is_meta"]=>
  string(3) "yes"
  ["field_width"]=>
  string(4) "full"
  ["options"]=>
  array(1) {
    [0]=>
    string(18) "This is a checkbox"
  }
  ["value"]=>
  string(18) "This is a checkbox"
  ["id"]=>
  string(18) "this_is_a_checkbox"
}
```

As you can see, the transformed array does not have a `label` key. This results in a `InvalidArgumentException` exception

## Acceptance Criteria

Donation receipts should not stall with a loader

Either
-   [ ] Make the `label` attribute required when creating/editing a custom form field

Or

-   [ ] Don't make `label` a required attribute in function `validateLineItem( $array )`

**Comment by ergosarapu** [comment]
Adding another related use case here.
We do not require donor name and therefore we set first name optional through `give_donation_form_required_fields` filter:
```
// Make First name field optional
unset($required_fields['give_first']);
```

When generating donor dashboard the array passed to `validateLineItem( $array )` contains empty 'value' key and this results
```
Uncaught Give\Framework\Exceptions\Primitives\InvalidArgumentException: Invalid receipt section line item.
Please provide valid line item id, label, and value. in .../plugins/give/src/Receipt/Section.php:150

Stack trace:
#0 .../plugins/give/src/Receipt/Section.php(89): Give\Receipt\Section->validateLineItem(Array)
#1 .../plugins/give/src/Receipt/DonationReceipt.php(60): Give\Receipt\Section->addLineItem(Array)
#2 .../plugins/give/src/Receipt/DonationReceipt.php(45): Give\Receipt\DonationReceipt->addDonorSection()
#3 .../plugins/give-recurring/src/DonorDashboard/Routes/SubscriptionsRoute.php(208): Give\Recei in .../plugins/give/src/Receipt/Section.php on line 150
```
Would be good if donor dashboard supports donations with empty donor name as well, similar to how missing billing address is handled.

**Comment by kjohnson** [comment]
@abirhoss thanks for the level of detail in the report! I'm moving this to the appropriate repository, which will close it here. You can track any progress updates via https://feedback.givewp.com/bug-reports/p/empty-label-attribute-in-custom-form-fields-causes-500-error-when-generating-rec
